[asset] Add useAssets hook to simplify preloading (#8928)
expo/expo: docs/pages/versions/unversioned/sdk/asset.md

<InstallSection packageName="expo-asset" />

# API

```js
import { Asset } from 'expo-asset';
```

## Hooks

### `useAssets`

```ts
const [assets, error] = useAssets([...]);
```

> Download and store one or more assets locally. After the assets are loaded, with [`Asset.loadAsync`](#assetloadasyncmodules), this hook returns a list of asset instances.

Reviewer: When we're describing a function, argument, class or whatever, we're talking about it in 3rd-person, so I think this would sound more natural 👇 

```suggestion
Downloads and stores one or more assets locally. After the assets are loaded, with [`Asset.loadAsync`](#assetloadasyncmodules), this hook returns a list of asset instances.
```

Author: Perfect, thanks 😄

> load one or many of asset modules to use in your app. After loading these assets, with [`Asset.loadAsync`](#assetloadasyncmodules), you can use these modules to use the downloaded asset.

Reviewer: Small suggestion, but generally the second sentence seems to be a little bit confusing 😅

Should we call them *modules*? Also it sounds like we still need to call `Asset.loadAsync` to use them 🤔 

```suggestion
Loads one or many of assets to use in your app. After loading these assets, with [`Asset.loadAsync`](#assetloadasyncmodules), you can use these modules to use the downloaded asset.
```

Author: Yes, I was doubting about calling them modules as well. Without is definitely better 👍

> Note, the assets are not "reloaded" when you dynamically change the asset list.

### Arguments

> - **assets (_number|number[]_)** -- One or multiple asset modules to load.

Reviewer: ```suggestion
- **assets (_number|number[]_)** -- One or multiple assets to load.
```

### Returns

- **assets (_Asset[]|undefined_)** -- A list of all loaded assets. If they aren't loaded yet, this value is `undefined`.
- **error (_Error|undefined_)** -- An error encountered when loading the assets.

### Example: hook

```tsx
function App() {
  const [assets] = useAssets([
    require('path/to/asset.jpg'),
    require('path/to/other.png'),
  ]);

  if (!assets) {
    return <AppLoading />;
  }

  return (
    <View>

>       <Text>Loaded {assets.length} assets!<Text>

Reviewer: ```suggestion
      <Text>Loaded {assets.length} assets!</Text>
```

Author: Yikes, thanks and sorry 😅
